### DISCUSSION TITLE
Dagster OSS vs Dagster+

### DISCUSSION THREAD

**Discussion by raayu83** [dagster-io/dagster#25313] [General]
Hello,

is there any webpage listing the differences between Dagster OSS and Dagster+?
While I'd really like to contribute to Dagster+ by requesting a commercial license, it would probably be easier to get adoption started with the OSS edition and possibly switch to Dagster+ later.
But for that I'd need to know if all essential features are supported.

Best regards

**Comment by garethbrickman** [comment]
Dagster+ will have all the features available in Open Source plus more. [This page](https://dagster.io/plus) describes those features. [This page in the official documentation](https://docs.dagster.io/dagster-plus/) gives a summary of the two Dagster+ deployment types and links to further reading for more technical-oriented details. 

For commercial discussions [please use this form](https://dagster.io/contact) to be in contact with our sales team.

**Reply by raayu83** [reply]
This still doesn‘t allow to compare the features of OSS vs Dagster+. This intransparency concerns me.

**Reply by raayu83** [reply]
@garethbrickman Thought about this again and decided against pursuing adoption of dagster due to the current intransparencies regarding pricing/features.

**Reply by garethbrickman** [reply]
Hey @raayu83, there is no "intransparency" intended here. Dagster+ will have all the features available in Open Source plus more. If there's any specific concern I can help address please let me know! Alternatively our Sales team can discuss your commercial needs with you privately.

**Reply by raayu83** [reply]
Hey @garethbrickman,

what I mean be intransparency is the following:
- There is no page highlighting Dagster+ only features vs open source features
- There are no examples on how many credits a typical ETL would consume, also see https://github.com/dagster-io/dagster/discussions/25316 regarding this

**Reply by garethbrickman** [reply]
There is information available for both those to an extent at: https://dagster.io/pricing

As the user below indicated, the table of features there shows Dagster+ features per plan type but not open source. I agree it can be worthwhile to have a column showing the open source features for comparison, I'll let our marketing team know that feedback. 

There is also a Pricing FAQ on that page which includes a description and example of credit usage measurement.

**Reply by raayu83** [reply]
Hi @garethbrickman ,
thanks for bringing this topic to the marketing team! It would really help a lot with transparency!

Regarding the Pricing FAQ, the example there is a really simple one. If my calculation the the linked discussion is correct, the credits needed can explode quickly, especially if you recalculate historic partitions on a daily basis. So I am kind of hoping my calculation is wrong...

**Comment by Radiergummi** [comment]
I'd like to cast my vote here, too. I have been searching for a page like these:
 - https://cloudsmith.com/pricing
 - https://tailscale.com/pricing
 - https://www.prefect.io/pricing

I'd definitely be able to justify the need to buy a license internally, but I don't even understand if I need one yet. This doesn't even have to include the actual amounts (even though I'd prefer that), but a feature breakdown is tremendously helpful.  
Don't take this as anything but feedback from a potential customer; I'm very interested in Dagster as a product.

**Reply by garethbrickman** [reply]
Dagster+ has such a page: https://dagster.io/pricing

But to be clear, you want to see the Dagster+ only features highlighted vs open source features?

**Comment by icarus4** [comment]
+1

Our team want to see the Dagster+ only vs. open source features. Just like Prefect does: https://www.prefect.io/cloud-vs-oss

The Dagster pricing page doesn't explain it well enough.

**Comment by daniel-clark** [comment]
For a discussion started in Oct '24, I'd expect more progress on this from Dagster's marketing team. The current lack of comparison between OSS and paid features gives a negative perception; is something being hidden? Being vague on purpose? Of course you assume the best, but the lack of clarity offers doubt.

**Reply by garethbrickman** [reply]
Hey @daniel-clark, I can assure you nothing is being hidden/made purposefully vague. We do have a website revamp project currently underway and this feedback has been surfaced to Dagster's marketing team. I've gone ahead and flagged again for them. In the meantime if you have any specific concerns please let me know!

**Comment by raayu83** [comment]
Hi @garethbrickman ,

one thing I'd like to ask for clarification on is credit usage.

For example, how many credits does the following scenario cost: 
- a job with a single task
- the job function is running a loop to call the task 10 times
- each call fills ten partitions (generating yesterdays partition + recalculating the nine days before)

Is this
- 1 credit
- 10 credits
- 100 credits

Or something else entirely?

**Reply by garethbrickman** [reply]
Your example terminology doesn't exactly line up with Dagster's paradigm, so I'll present some scenarios: 

Each job in Dagster contains one or more ops. A run of a Dagster job would be a single discrete run of those ops. 

So if you had a job with 1 op in it and you ran that job 10 times it would be 10 credits. 

If the "task" you're doing is looped 10 times inside of the op you would only need to run the job once, so it'd just be 1 credit. 

Dagster's [partitioning ops](https://docs.dagster.io/guides/build/partitions-and-backfills/partitioning-ops) means there would be a run for each partition, so credits would be the number of ops in the job x the number of partitions.

**Comment by raayu83** [comment]
Hi @garethbrickman ,

thanks your explanation!

Actually this system would make Dagster very expensive for us due to two reasons:
- We have quite a few jobs where we loop inside the job, not inside the op. In order to keep costs down, we would have to change them to loop inside the op instead. But I wonder if the jobs should really follow the structure of the pricing instead of being built the way that feels the most natural?
- We often recalculate past partitions. In some extreme cases we recalculate more than a hundred days during every run of a job, even though there is only very little data processed (less than a Megabyte per day). We could only justify implementing such cases in Dagster if recalculating partitions (as opposed to creating new partitions) was free.

**Reply by raayu83** [reply]
Regarding recalculating past partitions... could single_run backfill maybe help here?
Are multiple partitions handled in a single run billed as a single credit?
If so, can a normal schedule also trigger a single_run backfill? (e.g. if we wan to calculate yesterdays partition + recalculate the ten days before)

**Reply by garethbrickman** [reply]
A single-run backfill would still produce an asset materialization per partition- for commercial discussions around your specific needs [please use this form](https://dagster.io/contact) to be in contact with our sales team.

**Reply by raayu83** [reply]
Thanks, I've sent a message using the form.

**Comment by jox-gh** [comment]
It’s Jun ‘25 and I found this thread while searching for OSS vs Dagster+ features because it’s still not published anywhere.

**Reply by mindton** [reply]
I concur, having a transparent feature matrix is a fundamental requirement for OSS & Cloud comparison.
It provides needed clarity for those that are considering adopting Dagster.
Your product looks better than some competitors like Airflow, but this uncertainty/fog is discouraging.
It's logical you want people to use the cloud offering so it's sustainable and helps development too, but the fear of essentially losing features (i.e. introducing a new cloud-only feature and later deprecating a similar OSS one) is real and will drive people to alternatives even if subpar.
![image](https://github.com/user-attachments/assets/640932e4-c954-4d8a-a7a6-db5017c61416)
